tests/introspection/tests.py

```diff
@@ -184,14 +184,15 @@ def test_get_indexes_multicol(self):
         self.assertNotIn('first_name', indexes)
         self.assertIn('id', indexes)
 
-    @skipUnlessDBFeature('can_introspect_index_type')
     def test_get_constraints_indexes_types(self):
         with connection.cursor() as cursor:
             constraints = connection.introspection.get_constraints(cursor, Article._meta.db_table)
-        for key, val in constraints.items():
-            if val['index'] and not (val['primary_key'] or val['unique']):
-                self.assertIn('type', val)
-                self.assertEqual('btree', val['type'])
+        constraint = 'introspection_article_headline_pub_date_c749bad3_idx'
```

Comment:
To avoid the index name, how about this:
```
index = {}
for key, val in constraints.items():
    if val['columns'] == ['headline', 'pub_date']:
        index = val
        break
self.assertEqual(index['type'], 'btree')
```